apollo-runtime/src/main/java/com/apollographql/apollo/cache/normalized/RecordFieldJsonAdapter.java

```diff
@@ -84,6 +85,15 @@ private static void writeJsonValue(Object value, JsonWriter jsonWriter) throws I
         writeJsonValue(item, jsonWriter);
       }
       jsonWriter.endArray();
+    } else if (value instanceof LinkedHashMap) {
```

Comment:
I think we should use more generic type `Map` instead as we can easily change internal logic and use other  implementation of `Map` in future.